kiwi/builder/disk.py

```diff
@@ -653,24 +698,37 @@ def _build_and_map_disk_partitions(self):               # noqa: C901
                 filename=squashed_root_file.name,
                 exclude=[Defaults.get_shared_cache_location()]
             )
-            squashed_rootfs_mbsize = os.path.getsize(
-                squashed_root_file.name
-            ) / 1048576
+            squashed_rootfs_mbsize = int(
+                os.path.getsize(squashed_root_file.name) / 1048576
+            ) + Defaults.get_min_partition_mbytes()
             self.disk.create_root_readonly_partition(
-                int(squashed_rootfs_mbsize + 50)
+                squashed_rootfs_mbsize
             )
+            disksize_used_mbytes += squashed_rootfs_mbsize
+
+        if self.spare_part_mbsize and self.spare_part_is_last:
+            rootfs_mbsize = disksize_mbytes - disksize_used_mbytes - \
+                self.spare_part_mbsize - Defaults.get_min_partition_mbytes()
```

Comment:
Removing the `Defaults.get_min_partition_mbytes()` in this count is required? Looks like this could cause the spare_part size mismatch, even the mismatch I see has a different value...

Reply:
> There is no validation that the size of the spare partition is big enough to hold data of the mount point.

That applies to the volume setup as well if you give it a fixed size. At the sync level people will recognize this with a "no space left on device" error at build time. It would be very hard to pre-check this and in addition in depends on the filesystem if such a pre-size check can be successful. Thus we allow people to set the size and see if it fits

> Removing the Defaults.get_min_partition_mbytes() in this count is required?

This is the case when the spare_part is the last one. This leads to a reduction of the now required size calculation of the root partition. I intentionally substract the min partition size constraint to reach a minimum size available for the rest and to have a way to calculate with the inaccuracy between data, fs-overhead, partition-boundaries and such.

You are right the number value for the spare_part is not 100% accurate if spare_part is last. I could live with that. Thoughts ?

Reply:
> That applies to the volume setup as well if you give it a fixed size. At the sync level people will recognize this with a "no space left on device" error at build time. It would be very hard to pre-check this and in addition in depends on the filesystem

Well yes, you got a point here, probably we can't relay on making an accurate estimation of the minimal size required.

> You are right the number value for the spare_part is not 100% accurate if spare_part is last. I could live with that. Thoughts ?

I can certainly live with that, I was just wondering why the mismatch does not match the minimal partition default value.

All right then I think all is green here :+1:

Reply:
> Well yes, you got a point here, probably we can't relay on making an accurate estimation

I also did some tests with small and big spare_part's and I think you have a point here. I wanted to be at the very safe side with the size calculation. But I think we can be a bit less conservative. I've reduced the min size overhead which now leads to more accurate values.

spare_part="100M" ~ 102M
spare_part="5G" ~ 4984M

Let me know what your tests say

Thanks

Reply:
Yes, my test are also giving me way more accurate results now. :+1:

spare_part="200M" ~ 199M
spare_part="3G" ~ 2969Mb